pkg/artifactory/resource_artifactory_local_maven_repository.go

```diff
@@ -0,0 +1,92 @@
+package artifactory
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+var mavenLocalSchema = mergeSchema(baseLocalRepoSchema, map[string]*schema.Schema{
```

Comment:
please `closure` scope this

Reply:
Also, once you have closure scoped all these, the refactoring for generic function is very straightforward since the function arg is now scoped for all the usages.